crates/autopilot/src/run_loop.rs

```diff
@@ -64,12 +64,20 @@ pub struct RunLoop {
 
 impl RunLoop {
     pub async fn run_forever(self, update_interval: Duration) -> ! {
-        if let RunLoopMode::Unsynchronized = self.synchronization {
-            Maintenance::spawn_background_task(
-                self.maintenance.clone(),
-                self.eth.current_block().clone(),
-                update_interval,
-            );
+        match self.synchronization {
+            RunLoopMode::Unsynchronized => {
+                Maintenance::spawn_background_task(
+                    self.maintenance.clone(),
+                    self.eth.current_block().clone(),
+                    update_interval,
+                );
+            }
+            RunLoopMode::SyncToBlockchain => {
+                Maintenance::spawn_last_seen_block_watcher(
+                    self.eth.current_block().clone(),
+                    update_interval,
+                );
+            }
```

Comment:
nit: can we just have a single method on Maintenance `spawn_background_task` which takes the runloop mode and decides what it needs to span?